awx/main/scheduler/task_manager.py

```diff
@@ -510,13 +487,49 @@ def process_pending_tasks(self, pending_tasks):
                 self.start_task(task, None, task.get_jobs_fail_chain(), None)
                 continue
 
+            # Determine if ther is control capacity for the task
+            remaining_control_capacity = self.get_remaining_capacity('controlplane', capacity_type='control')
+            if remaining_control_capacity < settings.AWX_CONTROL_PLANE_TASK_IMPACT:
+                logger.debug(f"Skipping task {task.log_format} in pending, not enough capacity left on controlplane to control new tasks")
+                continue
+
+            control_instance = InstanceGroup.fit_task_to_most_remaining_capacity_instance(
+                task, self.graph['controlplane']['instances']
+            ) or InstanceGroup.find_largest_idle_instance(self.graph['controlplane']['instances'], capacity_type='control')
+            if not control_instance:
+                logger.debug(f"Skipping task {task.log_format} in pending, not enough capacity left on controlplane to control new tasks")
+                continue
+
+            task.controller_node = control_instance.hostname
+
             for rampart_group in preferred_instance_groups:
                 if task.capacity_type == 'execution' and rampart_group.is_container_group:
                     self.graph[rampart_group.name]['graph'].add_job(task)
                     self.start_task(task, rampart_group, task.get_jobs_fail_chain(), None)
                     found_acceptable_queue = True
                     break
 
+                if task.capacity_type != 'execution' and rampart_group.is_container_group:
+                    # This is a task that actually needs to run in the controlplane, like a project update for a container_group job
+                    if (
+                        not control_instance.remaining_capacity >= task.task_impact + settings.AWX_CONTROL_NODE_TASK_IMPACT
+                        and not control_instance.jobs_running == 0
+                    ):
+                        # As in other places, we accept an idle instance if the node with most capacity (the control node we already selected)
+                        # does not have enough capacity.
+                        logger.debug(f"Not enough control capacity on {control_instance} to run {task.log_format}")
+                        break
+                    task.execution_node = control_instance
+                    control_instance.remaining_capacity = max(
+                        0, control_instance.remaining_capacity - (task.task_impact + settings.AWX_CONTROL_PLANE_TASK_IMPACT)
+                    )
+                    control_instance.jobs_running += 1
+                    self.graph['controlplane']['graph'].add_job(task)
+                    execution_instance = self.real_instances[control_instance.hostname]
+                    self.start_task(task, 'controlplane', task.get_jobs_fail_chain(), execution_instance)
+                    found_acceptable_queue = True
+                    break
```

Comment:
Again, if you accept my prior comment about `capacity_type` being control, then the `if` condition here becomes unreachable and all of this block is deleted.